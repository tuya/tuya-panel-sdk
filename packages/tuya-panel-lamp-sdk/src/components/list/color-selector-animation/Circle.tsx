import React, { useRef, useEffect } from 'react';
import { StyleSheet, Animated } from 'react-native';
import { Utils } from 'tuya-panel-kit';
import { ICircle } from './interface';

const { convertX: cx } = Utils.RatioUtils;

const Circle: React.FC<ICircle> = ({ scaleValue, backgroundColor }) => {
  const fadeAnim = useRef(new Animated.Value(scaleValue)).current;
  useEffect(() => {
    Animated.spring(fadeAnim, {
      toValue: scaleValue,
    }).start();
  }, [scaleValue]);

  return (
    <Animated.View
      style={[
        styles.colorBtn,
        {
          backgroundColor,
          transform: [{ scale: fadeAnim }],
        },
      ]}
    />
  );
};

const styles = StyleSheet.create({
  colorBtn: {
    borderRadius: cx(14),
    height: cx(28),
    width: cx(28),
  },
});

export default Circle;
