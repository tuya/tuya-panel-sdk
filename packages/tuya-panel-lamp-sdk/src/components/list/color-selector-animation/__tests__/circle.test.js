import React from 'react';
import { shallow } from 'enzyme';
import Circle from '../Circle';

describe('Circle components', () => {
  it('basic render', () => {
    const wrapper = shallow(<Circle scaleValue="#333" backgroundColor="#fff" />);
    expect(wrapper).toMatchSnapshot();
  });
});
