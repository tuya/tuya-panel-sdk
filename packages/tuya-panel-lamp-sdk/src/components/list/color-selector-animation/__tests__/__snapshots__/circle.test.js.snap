// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle components basic render 1`] = `
<AnimatedComponent
  style={
    Array [
      Object {
        "borderRadius": 28,
        "height": 56,
        "width": 56,
      },
      undefined,
      Object {
        "backgroundColor": "#fff",
        "transform": Array [
          Object {
            "scale": "#3330",
          },
        ],
      },
    ]
  }
/>
`;
